### PR TITLE
feat(ingest): support writing configs to files

### PR DESCRIPTION
**Pull request by hsheth2**
Adds the `__DATAHUB_TO_FILE_` directive to our config loader.

Some sources (e.g. kafka, bigquery) require paths to files on a local file system. This doesn't work for UI ingestion, where the container needs to be totally self-sufficient.

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/tests/unit/config/write_to_file_directive.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+foo: bar
+nested:
+  hi: hello
+  __DATAHUB_TO_FILE_ssl_cert: |
```
Comment (jjoyce0510) [inline metadata-ingestion/tests/unit/config/write_to_file_directive.yml:4]: so how does the end user provide this? via the recipe or via environment variable?

Comment (hsheth2) [reply metadata-ingestion/tests/unit/config/write_to_file_directive.yml:4]: can be either

`__DATAHUB_TO_FILE_<key>: <value>` will get turned into `<key>: <path to file containing value>`

and value can be specified inline or using an env var/secret

File: metadata-ingestion/src/datahub/configuration/config_loader.py [hunk 2]
```diff
@@ -67,12 +68,40 @@ def list_referenced_env_variables(config: dict) -> Set[str]:
     return set([call[1][0] for call in calls])
 
 
+WRITE_TO_FILE_DIRECTIVE_PREFIX = "__DATAHUB_TO_FILE_"
```
Comment (asikowitz) [inline metadata-ingestion/src/datahub/configuration/config_loader.py:71]: Is this notion of "directive" a yaml thing or something we've introduce before?

**Review by asikowitz — APPROVED** [review]
I'm missing context on how we intend to use this feature. If this is just to help us run UI ingestions, this seems good. If we're going to suggest others do this, I could see it being confusing and would prefer generating these files in the relevant ConfigModel class rather than having users specify this behavior

File: metadata-ingestion/src/datahub/configuration/config_loader.py (hunk 2 above)
Comment (hsheth2) [reply metadata-ingestion/src/datahub/configuration/config_loader.py:71]: it's part of "datahub-flavored yaml", which is something I just made up :)

this change is mainly to make it easy to run UI ingestion - don't expect this to be a common way we do things broadly

**Pull request merged**